amxmodx/natives.cpp

```diff
@@ -112,7 +121,30 @@ int amxx_DynaCallback(int idx, AMX *amx, cell *params)
 		pDebugger->BeginExec();
 	}
 
-	err=amx_Exec(pNative->amx, &ret, pNative->func);
+	char perf_funcname[512];
+	CPluginMngr::CPlugin* perf_Plug = g_plugins.findPluginFast(pNative->amx);
+	amx_GetPublic(perf_Plug->getAMX(), pNative->func, perf_funcname);
+	const char* perf_plugname = perf_Plug->getName();
+
+	using std::chrono::high_resolution_clock;
+	using std::chrono::duration_cast;
+	using std::chrono::duration;
+	using std::chrono::milliseconds;
+
+	auto t1 = high_resolution_clock::now();
+
+	err = amx_Exec(pNative->amx, &ret, pNative->func);
+
+	auto t2 = high_resolution_clock::now();
+
+	auto ms_int = duration_cast<milliseconds>(t2 - t1);
+	auto ms_integer = (int)ms_int.count();
```

Comment:
This code is repeated almost verbatim four times. It'd be better to factor it out into a helper function and pass the amx_Exec in as a lambda (eg std::function).

Reply:
yes

Reply:
But i can't do it )